ocaml/xapi/cpuid_helpers.ml

```diff
@@ -200,22 +206,17 @@ let assert_vm_is_compatible ~__context ~vm ~host ?remote () =
         if vm_cpu_vendor <> host_cpu_vendor then
           fail "VM last booted on a host which had a CPU from a different vendor."
       end;
-      if List.mem_assoc cpu_info_features_key vm_cpu_info then begin
-        (* Check the VM was last booted on a CPU whose features are a subset of the features of this host's CPU. *)
-        let vm_cpu_features = Map_check.getf features vm_cpu_info in
-        debug "VM last booted on CPU with features %s; host CPUs have migration features %s"
-		(string_of_features vm_cpu_features) (string_of_features host_cpu_features');
-        let vm_cpu_features' =
-          vm_cpu_features
-          |> upgrade_features ~__context ~vm host_cpu_features'
-        in
-        if not (is_subset vm_cpu_features' host_cpu_features') then begin
-          debug "VM CPU features (%s) are not compatible with host CPU features (%s)\n" (string_of_features vm_cpu_features') (string_of_features host_cpu_features');
-          debug "Host is missing these CPU features required by the VM: %s" (string_of_features (diff vm_cpu_features' host_cpu_features'));
-          fail "VM last booted on a CPU with features this host's CPU does not have."
+      if List.mem_assoc cpu_info_policy_key vm_cpu_info then begin
```

Comment:
I think there has to be some upgrade path  from VMs that had the old v1 or v2 style features and do not have policies yet. Is there such a function somewhere?